src/vxworks/mod.rs

```diff
@@ -0,0 +1,2322 @@
+        pub condMutex: *mut ::pthread_mutex_t,
+        pub condAttr: ::pthread_condattr_t,
+        pub condSemName: [::c_char; PTHREAD_SHARED_SEM_NAME_MAX]
+    }
+
+    // b_pthread_rwlockattr_t.h
+    pub struct pthread_rwlockattr_t {
+        pub rwlockAttrStatus: ::c_int,
+        pub rwlockAttrMaxReaders: ::c_uint,
+    }
+
+    // b_pthread_rwlock_t.h
+    pub struct pthread_rwlock_t {
+        pub rwlockSemId: :: _Vx_SEM_ID,
+        pub rwlockReadersRefCount: ::c_int,
+        pub rwlockValid: ::c_int,
+        pub rwlockInitted: ::c_int,
+        pub rwlockAttr: ::pthread_rwlockattr_t,
+        pub rwlockName: [::c_char; PTHREAD_SHARED_SEM_NAME_MAX]
+    }
+
+    // b_struct_timeval.h
+    pub struct timeval {
+        pub tv_sec: ::time_t,
+        pub tv_usec: ::suseconds_t,
+    }
+
+    // socket.h
+    pub struct sockaddr {
+        pub sa_len    : ::c_uchar,
+        pub sa_family : sa_family_t,
+        pub sa_data   : [::c_char; 14],
+    }
+
+    // socket.h
+    pub struct sockaddr_storage {
+        pub ss_len     : ::c_uchar,
+        pub ss_family  : ::sa_family_t,
+        pub __ss_pad1  : [::c_char; _SS_PAD1SIZE],
+        pub __ss_align : ::int32_t,
+        // pub __ss_pad2  : [::c_char; _SS_PAD2SIZE],
+        pub __ss_pad2  : [::c_char; 32],
+        pub __ss_pad3  : [::c_char; 32],
+        pub __ss_pad4  : [::c_char; 32],
+        pub __ss_pad5  : [::c_char; 32],
+    }
+    pub struct iovec {
+        pub iov_base: *mut ::c_void,
+        pub iov_len: ::size_t,
+    }
+
+    // poll.h
+    pub struct pollfd {
+        pub fd      : ::c_int,
+        pub events  : ::c_short,
+        pub revents : ::c_short,
+    }
+
+    // dirent.h
+    pub struct dirent {
+        pub d_ino  : ::ino_t,
+        // pub d_name : [::c_char; (_PARM_NAME_MAX + 1)],
+        pub d_name : [::c_char; 32],
+        pub d_name1 : [::c_char; 32],
+        pub d_name2 : [::c_char; 32],
+        pub d_name3 : [::c_char; 32],
+        pub d_name4 : [::c_char; 32],
+        pub d_name5 : [::c_char; 32],
+        pub d_name6 : [::c_char; 32],
+        pub d_name7 : [::c_char; 32],
+    }
+
+    pub struct dirent64 {
+        pub d_ino    : ::ino_t,
+        pub d_off    : ::off64_t,
+        pub d_reclen : u16,
+        pub d_type   : u8,
+        // pub d_name   : [::c_char; 256],
+        pub d_name   : [::c_char; 32],
+        pub d_name1   : [::c_char; 32],
+        pub d_name2   : [::c_char; 32],
+        pub d_name3   : [::c_char; 32],
+        pub d_name4   : [::c_char; 32],
+        pub d_name5   : [::c_char; 32],
+        pub d_name6   : [::c_char; 32],
+        pub d_name7   : [::c_char; 32],
+    } // Doesn't seem like it exists anymore
+
+    // resource.h
+    pub struct rlimit { /* Is this really needed? Questionable ... */
+                           pub rlim_cur : ::size_t,
+                           pub rlim_max : ::size_t,
+    }
+
+    // stat.h
+    pub struct stat {
+                         pub st_dev       : ::dev_t,
```

Comment:
Something is off with the alignment here.

Reply:
Sure.